doc/adr/0017-mandatory-js-support.md

```diff
@@ -0,0 +1,47 @@
+# 17. Mandatory client-side JavaScript support
+
+Date: 2023-01-11
+
+## Status
+
+Accepted
+
+## Context
+
+All eServices should be able to integrate our service seamlessly.
+ADR 0016 determines this integration by specifying the usage of a JavaScript file that creates an iFrame for displaying the widget.
+However, the widget needs to execute / fetch logic, everytime it is loaded, to work correctly.
+We need to decide whether we want to have a server-side or client-side approach to prepare the widget.
+Most of the reasons and implications come down to the decision between using client-side JavaScript or not, because a client side approach would require the user to have JavaScript enabled.
+
+Client-side approach (JavaScript enabled):
+
+- benefits:
+  - client side event tracking is possible
+  - frugal use of data on the server (no need to store the constructed TCTokenURL on the server)
+  - additional room for client-side adjustments / use-cases
+- costs:
+  - potentially fewer users because we cannot enforce to enable JavaScript
+  - setting up a test environment requires more effort
+  - potential security vectors
+
+Server-side approach (JavaScript disabled):
+
+- benefits:
+  - smaller attack surface
+  - easier to debug and test, because we could test against static html and wouldn't need a JavaScript runtime
+  - potentially more users because we do not rely on enabled JavaScript
+- costs:
+  - storage of unnecessary data on the server
+  - difficult client-side event tracking
```

Comment:
what does it mean client-side event tracking is "difficult"? devs could still add event payload in the request body in the form (e.g. submitting form), but I think it's more preference.

https://developer.mozilla.org/en-US/docs/Learn/Forms/Sending_and_retrieving_form_data